app/src/main/java/org/mozilla/fenix/tabtray/TabTrayView.kt

```diff
@@ -258,7 +258,29 @@ class TabTrayView(
 
         adjustNewTabButtonsForNormalMode()
 
+        @Suppress("ComplexCondition")
         if (
+            view.context.settings().showGridViewInTabsSettings &&
+            view.context.settings().shouldShowGridViewBanner &&
+            view.context.settings().canShowCfr &&
+            tabs.size >= TAB_COUNT_SHOW_CFR
```

Comment:
Does grid view have the same tabs size requirement?

Reply:
Ah, I forgot to add that we are using the same CFR requirements as the auto-close except that we are gonna show the grid view banner first.